test/e2e/framework/service_util.go

```diff
@@ -525,6 +528,44 @@ func (j *ServiceTestJig) newRCTemplate(namespace string) *v1.ReplicationControll
 	return rc
 }
 
+func (j *ServiceTestJig) AddRCAntiAffinity(rc *v1.ReplicationController) {
```

Comment:
Minor nit, this tweak doesn't need to be attached to the jig. It could live along with the upgrade test that needs it.